src/isomorphic/classic/types/__tests__/ReactPropTypesProduction-test.js

```diff
@@ -0,0 +1,286 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+
+describe('ReactPropTypesProduction', function() {
+  var oldProcess;
+  var PropTypes;
+  var React;
+  var ReactTestUtils;
+
+  beforeEach(function() {
+    __DEV__ = false;
+    oldProcess = process;
+    global.process = {env: {NODE_ENV: 'production'}};
+
+    jest.resetModuleRegistry();
+    PropTypes = require('ReactPropTypes');
+    React = require('React');
+    ReactTestUtils = require('ReactTestUtils');
+  });
+
+  afterEach(function() {
+    __DEV__ = true;
+    global.process = oldProcess;
+  });
+
+  function typeCheckPass(declaration, value) {
+    var props = {testProp: value};
+    declaration(
+      props,
+      'testProp',
+      'testComponent',
+      'prop'
+    );
+    const expectedCount = __DEV__ ? 0 : 1;
+    expect(console.error.calls.count()).toBe(expectedCount);
+    if (!__DEV__) {
```

Comment:
I think it is unnecessary. Our tests are set up in a way that any unexpected warning would fail a test. So if it showed up anywhere, we’d notice thanks to existing DEV mode tests. Let’s make these tests production-only. This simplifies the logic too.